torch/_prims/__init__.py

```diff
@@ -1335,25 +1355,12 @@ def _collapse_view_meta(a: TensorLikeType, start: int, end: int) -> TensorLikeTy
         msg = "Attempting to view a collapsed tensor, but no such view exists!"
         raise ValueError(msg)
 
-    if new_strides is None:
-        return a.view(new_shape)
-    else:
-        return a.as_strided(new_shape, new_strides, a.storage_offset())
+    assert new_strides is not None
```

Comment:
Let's just raise within `collapse_view_helper` and never return `None`.

Reply:
`_collapse_view_helper` is used by `_refs.flatten` which returns a copy when `_collapse_view_helper` returns `None`.

Reply:
oof. ok